apps/aecore/lib/aecore/util/header.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Aecore.Util.Header do
+  @moduledoc """
+  Header utilities
+  """
+  alias Aecore.Chain.KeyHeader
+  alias Aecore.Chain.MicroHeader
+
+  @spec hash(KeyHeader.t() | MicroHeader.t()) :: binary()
+  def hash(header) do
+    case header do
```

Comment:
This case can be removed by specifying two clauses in the hash function, eg:
```
def hash(%KeyHeader{}) do
...
end
def hash(%MicroHeader()) do
...
end
...
```
Or by writing:
 ``` 
def hash(header) when header.__struct__ in [KeyHeader, MicroHeader] do
    header.__struct__.hash(header)
end
```

Reply:
Code reduction is always nice, will change it 😆 I dislike the second one